ext/hermann/hermann_lib.c

```diff
@@ -359,22 +353,48 @@ static void consumer_consume_stop_callback(void *ptr) {
 }
 #endif
 
+void *consumer_recv_msg(void *ptr)
+{
+	rd_kafka_message_t *ret;
+	HermannInstanceConfig *consumerConfig = (HermannInstanceConfig *) ptr;
+
+	ret = rd_kafka_consume(consumerConfig->rkt, consumerConfig->partition, 100);
+
+	if ( ret == NULL ) {
+		if ( errno != ETIMEDOUT )
+			fprintf(stderr, "%% Error: %s\n", rd_kafka_err2str( rd_kafka_errno2err(errno)));
+	}
+
+	return (void *) ret;
+}
+
 /**
  * Loop on a timeout to receive messages from Kafka.  When the consumer_consume_stop_callback is invoked by Ruby,
  * we'll break out of our loop and return.
  */
 void consumer_consume_loop(HermannInstanceConfig* consumerConfig) {
-
+	rd_kafka_message_t *msg;
 	TRACER("\n");
 
 	while (consumerConfig->run) {
-		if (rd_kafka_consume_callback(consumerConfig->rkt, consumerConfig->partition,
-										  1000/*timeout*/,
-										  msg_consume,
-										  consumerConfig) < 0) {
-			fprintf(stderr, "%% Error: %s\n", rd_kafka_err2str( rd_kafka_errno2err(errno)));
-		}
+#ifdef RB_THREAD_BLOCKING_REGION
+		msg = rb_thread_blocking_region(consumer_recv_msg,
+				consumerConfig,
+				consumer_consume_stop_callback,
+				consumerConfig);
+#elif HAVE_RB_THREAD_CALL_WITHOUT_GVL
+		msg = rb_thread_call_without_gvl(consumer_recv_msg,
+				consumerConfig,
+				consumer_consume_stop_callback,
+				consumerConfig);
+#else
+		msg = consumer_recv_msg(consumerConfig);
+#endif
 
+		if ( msg ) {
```

Comment:
This block concerns me, not because it's bad, but because it tells me that we were probably leaking before on the MRI consumer :/